lib/resty/healthcheck.lua

```diff
@@ -1505,6 +1506,25 @@ function _M.get_target_list(name, shm_name)
     return true
   end)
 
+  for _, target in ipairs(self.targets) do
+    target.counter = {
+      success = 0,
+      http_failure = 0,
+      tcp_failure = 0,
+      timeout_failure = 0,
+    }
```

Comment:
Why initialize to 0? is it better that target.counter is nil when error?

Reply:
The counter shm may be non-exist when the checker just starts up, so chances are that locking_target gets nothing.